### PR TITLE
feat(database): support `link` column type

### PR DESCRIPTION
**Pull request by zqran**
close: #2667

https://github.com/toeverything/blocksuite/assets/15389209/b77cd85f-90ca-4db1-b4b6-1cf5c359adec

### CODE REVIEW

File: packages/blocks/src/database-block/table/components/column-type/link.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+        value = this._initYText(this.value);
+      } else {
+        value = this.value;
+      }
+    }
+
+    this.vEditor = new VEditor(value, {
+      active: () => activeEditorManager.isActive(this),
+    });
+    setupVirgoScroll(this.column.page, this.vEditor);
```
Comment (zzj3720) [inline packages/blocks/src/database-block/table/components/column-type/link.ts:110]: Can we not depend on Virgo here

File: packages/blocks/src/database-block/table/components/selection/selection.ts [hunk 2]
```diff
@@ -96,11 +96,12 @@ export class DatabaseSelectionView extends WithDisposable(ShadowlessElement) {
...
     this._disposables.add({
       dispose: this.eventDispatcher.add('click', context => {
-        const target = context.get('pointerState').event.target;
+        const event = context.get('pointerState').event;
+        const target = event.target;
         if (target instanceof Element && this.isCurrentDatabase(target)) {
           const cell = target.closest('affine-database-cell-container');
           if (cell) {
-            cell?.cell?.enterEditMode();
+            cell?.cell?.enterEditMode(event);
```
Comment (zzj3720) [inline packages/blocks/src/database-block/table/components/selection/selection.ts:104]: enterEditMode may be called without an `event` in future, Can we not depend an `event` here

Comment (zqran) [reply packages/blocks/src/database-block/table/components/selection/selection.ts:104]: Here I want to trigger the click of the link when clicking the external area of the link, so I need to get the relevant elements.

image

---

Related code:

https://github.com/toeverything/blocksuite/pull/3187/files#diff-7f1b6d33e4075b6a3dc56f7a12e6d0bfca7122157175c37bfe3eaad36c347ec7R127-R135

Comment (zqran) [reply packages/blocks/src/database-block/table/components/selection/selection.ts:104]: Get!

File: packages/blocks/src/database-block/table/components/column-type/link.ts (hunk 1 above)
Comment (zqran) [reply packages/blocks/src/database-block/table/components/column-type/link.ts]: ok

**Pull request merged**